simple_history/tests/tests/test_models.py

```diff
@@ -1413,3 +1416,61 @@ def test_history_user_does_not_exist(self):
 
         self.assertEqual(user_id, instance.history.first().history_user_id)
         self.assertIsNone(instance.history.first().history_user)
+
+
+class RelatedNameTest(TestCase):
+    def setUp(self):
+        self.user_one = get_user_model().objects.create(
+            username="username_one", email="first@user.com", password="top_secret"
+        )
+        self.user_two = get_user_model().objects.create(
+            username="username_two", email="second@user.com", password="top_secret"
+        )
+
+        self.one = Street(name="Test Street")
+        self.one._history_user = self.user_one
+        self.one.save()
+
+        self.two = Street(name="Sesame Street")
+        self.two._history_user = self.user_two
+        self.two.save()
+
+        self.one.name = "ABC Street"
+        self.one._history_user = self.user_two
+        self.one.save()
+
+    def test_relation(self):
+        self.assertEqual(self.one.history.count(), 2)
+        self.assertEqual(self.two.history.count(), 1)
+
+    def test_filter(self):
+        self.assertEqual(
+            Street.objects.filter(history__history_user=self.user_one.pk).count(), 1
+        )
+        self.assertEqual(
+            Street.objects.filter(history__history_user=self.user_two.pk).count(), 2
+        )
+
+    def test_name_equals_manager(self):
+        with self.assertRaises(RelatedNameConflictError):
+            register(Place, manager_name="history", related_name="history")
+
+    def test_deletion(self):
+        self.two.delete()
+
+        self.assertEqual(Street.log.filter(history_relation=2).count(), 2)
+        self.assertEqual(Street.log.count(), 4)
+
+    def test_revert(self):
+        id = self.one.pk
```

Comment:
this test is a little confusing to me. Was hoping for something that would tell me that all of the previous historical records are still related to the reverted object, so:
```
id = self.one.id 
self.one.delete()

self.assertEqual(0, Street.objects.filter(pk=id).count())

old = Street.history.filter(id=id).first()
old.history_object.save()
self.assertEqual(4, old.history.count())

Reply:
Essentially just want to make sure that the same key is reused and all the foreign keys still link to the same base object

Reply:
I now test whether the object behaves as before and whether the relations still exist